tox.ini

```diff
@@ -26,34 +21,28 @@ commands =
 [testenv:tests_pytest_py3]
 changedir = {toxinidir}/packages/testing
 package = editable
-extras =
-    test
-    lint
 commands =
     pytest \
         -n auto --maxprocesses 6 \
         --basetemp="{temp_dir}/pytest" \
         src
 
 [testenv:tests_pytest_pypy3]
+dependency_groups =
+    test
 passenv =
     PYPY_GC_MAX
     PYPY_GC_MIN
 # run inside the package directory so pytest picks up packages/testing/pyproject.toml
 changedir = {toxinidir}/packages/testing
 package = editable
-extras =
-    test
-    lint
```

Comment:
Removing the `lint` extras caused the `gentest` unit tests to fail as `gentest` requires `ruff` (it formats the generated files for convenience).

Reply:
Finding this bug through tighter (and only listing the expected) dependency groups now makes me lean to keeping it more explicit everywhere. But this `ruff` dep issue is somewhat of an edge case.

Reply:
`uvx` allows ephemeral venvs and should allow running a tool on a package without polluting the package. If it worked, it seems like the cleanest way to go. Had a play, documented it here:
- https://github.com/ethereum/execution-specs/issues/1716